tasks/files-to-granules/index.js

```diff
@@ -39,12 +39,21 @@ async function fileObjectFromS3URI(s3URI) {
  * @param {Array<string>} inputFiles - list of s3 files to add to the inputgranules
  * @param {Array<Object>} inputGranules - an array of the granules
  * @param {string} regex - regex needed to extract granuleId from filenames
+ * @param {boolean} matchFilesWithProducerGranuleId -
+ *  If true, match files to granules using producerGranuleId. Else, granuleId.
  * @returns {Object} inputGranules with updated file lists
  */
-async function mergeInputFilesWithInputGranules(inputFiles, inputGranules, regex) {
+async function mergeInputFilesWithInputGranules(
+  inputFiles,
+  inputGranules,
+  regex,
+  matchFilesWithProducerGranuleId
+) {
   // create hash list of the granules
   // and a list of files
-  const granulesHash = keyBy(inputGranules, 'granuleId');
+  const granulesHash = matchFilesWithProducerGranuleId ?
```

Comment:
So.... what happens if the granules don't have a producer granuleId and the config flag is set? 

It looks like it failes on L71 (logged L73) with a not-useful error.       Do we want the task to fail in that instance?  If so, should we do some validation up front? 

If not, the error should definitely be improved given we can no longer assume `fileGranuleId` is a valid hash key at L71

We should have a test case documenting intent as well. 

Nit: Consider renaming 'fileGranuleId' as it's no longer necessarily a fileGranuleId.     "getGranuleId" is also now *very* loosely getting a granueld, because it's a producerGranuleId *or* GranuleId

Nit Nit: jsdoc types or TS would help a lot with clarity here.

Reply:
Also https://github.com/nasa/cumulus/pull/3990/commits/002d7ce2b0e98aede7a95234313ec7ac7ea8d2f7